dlc/src/main/scala/org/bitcoins/dlc/BinaryOutcomeDLCWithSelf.scala

```diff
@@ -183,21 +184,48 @@ case class BinaryOutcomeDLCWithSelf(
   }
 
   def createFundingTransaction: Future[Transaction] = {
-    val output: TransactionOutput =
-      TransactionOutput(totalInput, P2WSHWitnessSPKV0(fundingSPK))
-    val localChange =
-      TransactionOutput(Satoshis(localFunding) - localInput, localChangeSPK)
-    val remoteChange =
-      TransactionOutput(Satoshis(remoteFunding) - remoteInput, remoteChangeSPK)
-
-    val outputs: Vector[TransactionOutput] =
-      Vector(output, localChange, remoteChange)
-    val txBuilderF: Future[BitcoinTxBuilder] =
-      BitcoinTxBuilder(outputs, fundingUtxos, feeRate, emptyChangeSPK, network)
-
-    txBuilderF.flatMap { txBuilder =>
-      subtractFeeFromOutputsAndSign(txBuilder,
-                                    Vector(localChangeSPK, remoteChangeSPK))
+
+    val emptyOutPoint =
+      TransactionOutPoint(DoubleSha256DigestBE.empty, UInt32.zero)
+    val dummySpendingInfo = P2WSHV0SpendingInfo(
+      outPoint = emptyOutPoint,
+      amount = totalInput * 2,
+      scriptPubKey = P2WSHWitnessSPKV0(fundingSPK),
+      signers = Vector(fundingLocalPrivKey, fundingRemotePrivKey),
+      hashType = HashType.sigHashAll,
+      scriptWitness = P2WSHWitnessV0(fundingSPK),
+      conditionalPath = ConditionalPath.NoConditionsLeft
+    )
+    val cetWinLocalF = createCETWinLocal(dummySpendingInfo)
+    val cetFeeF = cetWinLocalF.map { tx =>
+      feeRate.calc(tx._1)
+    }
+
+    cetFeeF.flatMap { cetFee =>
+      val halfCetFee = Satoshis(cetFee.satoshis.toLong / 2)
+
+      val output: TransactionOutput =
+        TransactionOutput(totalInput + cetFee, P2WSHWitnessSPKV0(fundingSPK))
```

Comment:
It would be good to have some comment to narrate what is going on in here. Here is how I understand it:

>We need to commit to the fee to execute the `cet` during the construction of the funding transaction. Our approach at the moment is to estimate the currenty blockchain's fee rate, and then approximate how many bytes the `cetWinLocal` transaction will be (why did we choose this one?). We then add that amount to the `fundingOutput` so it can be used for fees later.

Is this right?

Reply:
yup, with the only additional comment that this is all done so that the CETs output the amount specified in the contract and not less.